server/chat-commands/info.js

```diff
@@ -2398,33 +2398,29 @@ const commands = {
 	code(target, room, user) {
 		if (!target) return this.parse('/help code');
 		if (!this.canTalk()) return;
-		if (target.startsWith('\n')) target = target.slice(1);
 		if (target.length >= 8192) return this.errorReply("Your code must be under 8192 characters long!");
-		const separator = '\n';
-		if (target.includes(separator) || target.length > 150) {
-			const params = target.split(separator);
-			let output = [];
-			let cutoff = 3;
-			for (const param of params) {
-				if (output.length < 2 && param.length > 80) cutoff = 2;
-				output.push(Chat.escapeHTML(param));
-			}
-			let code;
-			if (output.length > cutoff) {
-				code = `<div class="chat"><details class="readmore code" style="white-space: pre-wrap; display: table; tab-size: 3"><summary>${output.slice(0, cutoff).join('<br />')}</summary>${output.slice(cutoff).join('<br />')}</details></div>`;
-			} else {
-				code = `<div class="chat"><code style="white-space: pre-wrap; display: table; tab-size: 3">${output.join('<br />')}</code></div>`;
-			}
 
-			if (!this.canBroadcast(true, '!code')) return;
-			if (this.broadcastMessage && !this.can('broadcast', null, room)) return false;
-
-			if (!this.runBroadcast(true, '!code')) return;
+		const params = target.split('\n');
+		let output = [];
+		let cutoff = 3;
+		for (const param of params) {
+			if (output.length < 2 && param.length > 80) cutoff = 2;
+			output.push(Chat.escapeHTML(param));
+		}
 
-			this.sendReplyBox(code);
+		let code;
+		if (output.length > cutoff) {
+			code = `<div class="chat"><details class="readmore code" style="white-space: pre-wrap; display: table; tab-size: 3"><summary>${output.slice(0, cutoff).join('<br />')}</summary>${output.slice(cutoff).join('<br />')}</details></div>`;
 		} else {
-			return this.errorReply("You can simply use ``[code]`` for code messages that are only one line.");
+			code = `<div class="chat"><code style="white-space: pre-wrap; display: table; tab-size: 3">${output.join('<br />')}</code></div>`;
 		}
+
+		if (!this.canBroadcast(true, '!code')) return;
+		if (this.broadcastMessage && !this.can('broadcast', null, room)) return false;
+
+		if (!this.runBroadcast(true, '!code')) return;
```

Comment:
I'm not sure if you need the first or second broadcast checks, `runBroadcast` will check `canBroadcast`, which itself will check `this.can('broadcast, null, this.room)`.

Reply:
I'm also confused what those first two checks are for.

Reply:
Does this check need to happen this late in the function either?

Reply:
@Kaiepi No, you can move it up if you'd like.